waiter/src/waiter/main.clj

```diff
@@ -64,15 +65,47 @@
   [handler]
   (fn [{:keys [body] :as request}]
     (let [{:keys [internal-protocol] :as response} (handler request)]
-      (when (and (instance? ServletInputStream body)
+      (if (and (instance? ServletInputStream body)
                  (not (.isFinished ^ServletInputStream body))
                  (not (instance? ManyToManyChannel response))
                  (not (hu/http2? internal-protocol)))
-        (try
-          (slurp body)
-          (catch IOException e
-            (log/error e "Unable to consume request stream"))))
-      response)))
+        (let [response-ch (async/promise-chan)
+              input-stream ^ServletInputStream body
+              bytes-counter-atom (atom 0)
+              throughput-meter-global (metrics/waiter-meter "streaming" "request-bytes")
+              throughput-iterations-meter-global (metrics/waiter-meter "streaming" "request-iterations")
+              complete-request-streaming (fn complete-request-streaming [throwable]
+                                           (if throwable
+                                             (log/error throwable "error after consuming" @bytes-counter-atom "bytes from request stream")
+                                             (log/info "successfully consumed" @bytes-counter-atom "bytes from request stream"))
```

Comment:
How do we know which request these log lines are talking about? Don't we need the cid?

Reply:
Will add the CID.